Sources/OpenAPI/OpenApiProvider.swift

```diff
@@ -0,0 +1,184 @@
+        }
+        set {
+            guard let key = Key.init(string) else { return }; #warning("This should probably throw an error or something")
+            self[key] = newValue
+        }
+    }
+}
+
+public struct OpenAPIKeyString: RegexableString {
+    public static var regex: String = #"^[a-zA-Z0-9\.\-_]+$"#
+    
+    public var value: String
+}
+
+public struct OpenAPIComponents: Content {
+    public let schemas: [OpenAPIKeyString: OpenAPISchema]?
+    public let responses: [OpenAPIKeyString: OpenAPIResponse]?
+    public let parameters: [OpenAPIKeyString: OpenAPIParameter]?
+    public let examples: [OpenAPIKeyString: OpenAPIExample]?
+    public let requestBodies: [OpenAPIKeyString: OpenAPIExample]?
+    public let headers: [OpenAPIKeyString: OpenAPIHeader]?
+    public let securitySchemes: [OpenAPIKeyString: OpenAPISecurityScheme]?
+    public let links: [OpenAPIKeyString: OpenAPILink]?
+    public let callbacks: [OpenAPIKeyString: OpenAPICallback]?
+}
+
+public struct OpenAPIPathString: RegexableString {
+    public static var regex: String = #"\/.*"#
+    
+    public var value: String
+}
+
+public struct OpenAPIPath: Content {
+    public let summary: String?
+    public let description: String?
+    public let get: OpenAPIOperation?
+    public let put: OpenAPIOperation?
+    public let post: OpenAPIOperation?
+    public let delete: OpenAPIOperation?
+    public let options: OpenAPIOperation?
+    public let head: OpenAPIOperation?
+    public let patch: OpenAPIOperation?
+    public let trace: OpenAPIOperation?
+    public let servers: [OpenAPIServer]?
+    public let parameters: [OpenAPIParameter]?
+}
+
+public struct OpenAPIOperation: Content {
+    public let tags: [String]?
+    public let summary: String?
+    public let description: String?
+    public let externalDocs: OpenAPIExternalDocs?
+    public let operationId: String
+    public let parameters: [OpenAPIParameter]?
+    public let requestBody: OpenAPIRequestBody?
+    public let responses: OpenAPIResponses
+    public let callbacks: [String: OpenAPICallback]?
+    public let deprecated: Bool?
+    public let security: [OpenAPISecurityRequirement]?
+    public let servers: [OpenAPIServer]?
+}
+
+public struct OpenAPIExternalDocs: Content {
+    public let description: String?
+    public let url: URL
+}
+
+public struct OpenAPIParameter: Content {
+    public let name: String
+    public let `in`: OpenAPIParameterLocation
+    public let description: String?
+    public private(set) var required: Bool; #warning(#"TODO: If the parameter location is "path", this property is REQUIRED and its value MUST be true. Otherwise, the property MAY be included and its default value is false."#)
+    public let deprecated: Bool?
+    public let allowEmptyValue: Bool
+}
+
+public enum OpenAPIParameterLocation: String, Content {
+    case query, header, path, cookie
+}
+
+
+public struct OpenAPISpec: Content {
+    public let openapi: String = "3.0.0"
```

Comment:
```suggestion
    public let openAPI: String = "3.0.0"
```
If this is all lowercase in the JSON, ignore me.

Reply:
Yup, JSON has it all lowercase :P